tournamentcontrol/competition/tests/test_utils.py

```diff
@@ -124,3 +124,98 @@ def test_sum_dict_integer(self):
 
     def test_revpow(self):
         self.assertEqual(utils.revpow(2**4, 2), 4)
+
+
+class StageGroupPositionTests(TestCase):
+    """
+    Tests for the stage_group_position function.
+    """
+
+    def test_stage_group_position_with_no_pools_raises_index_error(self):
+        """
+        Test that stage_group_position raises IndexError when trying to access
+        a pool that doesn't exist on a stage.
+        """
+        from tournamentcontrol.competition.tests.factories import (
+            DivisionFactory,
+            StageFactory,
+        )
+
+        # Create a division with multiple stages so we can reference one
+        division = DivisionFactory()
+        stage1 = StageFactory(division=division, order=1)
+        stage2 = StageFactory(division=division, order=2)
```

Comment:
This is not how to use factory-boy - there should be a create call.

Check other examples of these factories.